compiler/src/iree/compiler/Codegen/Common/PropagateReshapesByExpansion.cpp

```diff
@@ -312,6 +312,22 @@ void PropagateReshapesByExpansionPass::runOnOperation() {
   }
 
   RewritePatternSet bubbleExpandShapePatterns(context);
+  // Add patterns to do some additional cleanup (on top of canonicalizations
+  // that can be done later) of reshape ops.
+  // TODO (nirvedhmeshram, Max191) : The canonicalization patterns need to run
```

Comment:
Do we create a section and run the canonicalization patterns separately?

The populate order does impact the priority. We'd prefer using benefits, but they are not available for canonicalization patterns, I think. So I'm wondering if we can run them separately.

Reply:
The problem is that then we wont have the patterns run after the reshape patterns which is something we also want.

Reply:
Yeah this won't work for the reason Hanhan is saying. The order in which patterns are populated has no relation with the application order. Canonicalization patterns always have a benefit of 1 I believe, so we can just give the reshape patterns a lower benefit if that's an issue.